Tests/Shared/VideoTranscodeStatusTests.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  VideoTranscodeStatusTests.swift
+//  VimeoNetworking-iOSTests
+//
+//  Created by Nicole Lehrer on 12/10/19.
+//  Copyright © 2019 Vimeo. All rights reserved.
+//
+
+import XCTest
+@testable import VimeoNetworking
+
+class VideoTranscodeStatusTests: XCTestCase {
+    
+    let mockWithAllValidData = Data("""
+    {"state": "active", "progress": 50, "time_left": 5}
+    """.utf8)
+    
+    let mockWithInvalidState = Data("""
+    {"state": "bogusState", "progress": 50, "time_left": 5}
+    """.utf8)
+    
+    let mockWithInvalidProgress = Data("""
+    {"state": "active", "progress": 2000000000000000000000, "time_left": 5}
+    """.utf8)
+
+    let mockWithInvalidTimeLeft = Data("""
+       {"state": "active", "progress": 50, "time_left": ""}
+       """.utf8)
+    
+    // MARK: - Custom Decoder Tests
+
+    func testCustomDecoder_decodesValidData() {
+        do {
+            let statusObject = try JSONDecoder().decode(VideoTranscodeStatus.self, from: mockWithAllValidData)
+            XCTAssertTrue(statusObject.state == .active)
+            XCTAssertTrue(statusObject.progress == 50)
+            XCTAssertTrue(statusObject.timeLeft == 5)
+        }
+        catch let error {
+            XCTFail("Custom decoding failed with unexpected \(error)")
+        }
+    }
+    
+    func testCustomDecoder_returnsDataCorruptedError_onInvalidStatus_asEmptyValue() {
+        do {
+            let _ = try JSONDecoder().decode(VideoTranscodeStatus.self, from: mockWithInvalidState)
+            XCTFail("Custom decoding should have failed")
+        }
+        catch let DecodingError.dataCorrupted(context) {
+            XCTAssertNotNil(context)
+        }
+        catch {
+            XCTFail("Unexpected failure")
+        }
+     }
+    
+    func testCustomDecoder_returnsDataCorruptedError_onInvalidProgress_valueLargerThanIntSize() {
+       do {
+           let _ = try JSONDecoder().decode(VideoTranscodeStatus.self, from: mockWithInvalidProgress)
+           XCTFail("Custom decoding should have failed")
+       }
+       catch let DecodingError.dataCorrupted(context) {
+           XCTAssertNotNil(context)
+       }
+       catch {
+           XCTFail("Unexpected failure")
+       }
+    }
+    
+    func testCustomDecoder_returnsTypeMismatchError_onInvalidTimeLeft_valueEmptyString() {
+       do {
+           let _ = try JSONDecoder().decode(VideoTranscodeStatus.self, from: mockWithInvalidTimeLeft)
+           XCTFail("Custom decoding should have failed")
+       }
+       catch let DecodingError.typeMismatch(context) {
+           XCTAssertNotNil(context)
+       }
+       catch {
+           XCTFail("Unexpected failure")
+       }
+    }
+    
+    // MARK: - TranscodeStatusEndpointType Tests
+
+    func testTranscodeStatusEndpointType_returnsCorrectURL() {
```

Comment:
I know this may seem like overkill but how about moving this to a separate testfile since it is testing a different component? This will make it easier when we move models and their tests out of this repo.